lib/ansible/plugins/become/sudo.py

```diff
@@ -94,7 +97,20 @@ def build_become_command(self, cmd, shell):
         if self.get_option('become_pass'):
             self.prompt = '[sudo via ansible, key=%s] password:' % self._id
             if flags:  # this could be simplified, but kept as is for now for backwards string matching
-                flags = flags.replace('-n', '')
+                reflag = []
+                for flag in shlex.split(flags):
+                    if flag in ('-n', '--non-interactive'):
+                        continue
+                    elif not flag.startswith('--'):
+                        # handle -XnxxX flags only
+                        flag = re.sub(r'^(-\w*)n(\w*.*)', r'\1\2', flag)
```

Comment:
There appears to be a problem here on Py2.6 in that it's turning the options into:

```
-\x00\x00\x00H\x00\x00\x00
```

Example of `reflag`:

```
['-\x00\x00\x00H\x00\x00\x00', '\x00\x00\x00-\x00\x00\x00S\x00\x00\x00', '\x00\x00\x00-\x00\x00\x00n\x00\x00\x00']
```

Reply:
pushed flags=re.U to fix

Reply:
Python 2.6's `re.sub` doesn't accept `flags` -- but `re.compile` does.

Reply:
had just figured that out in ssh_debug backport, thanks!